websocket/README.md

```diff
@@ -0,0 +1,73 @@
+## Wing WebSocket Support
+
+This library allows you tu create WebSocket usiing Wing.
+
+## Installation
+
+Use `npm` to iinstall this library:
+
+```sh
+npm i @winglibs/websocket
+```
+
+## Bring it
+
+The example above shows us how we can broadcast a message to any connection assigned to the WebSocket.
+
+```js
+bring cloud;
+bring ex;
+
+let tb = new ex.DynamodbTable(
+  name: "WebSocketTable",
+  hashKey: "connectionId",
+  attributeDefinitions: {
+    "connectionId": "S",
+  },
+);
+
+let wb = new websocket.WebSocket(name: "MyWebSocket") as "my-websocket";
+
+wb.onConnect(inflight(id: str): void => {
+  tb.putItem({
+    item: {
+      "connectionId": id
+    }
+  });
+});
+
+wb.onDisconnect(inflight(id: str): void => {
+  tb.deleteItem({
+    key: {
+      "connectionId": id
+    }
+  });
+});
+
+wb.onMessage(inflight (id: str, body: str): void => {
+  let connections = tb.scan();
+  for item in connections.items {
+    wb.sendMessage(str.fromJson(item.get("connectionId")), body);
+  }
+});
+
+wb.initialize();
```

Comment:
Is this necessary?

Reply:
I think it is required because we are not using `sim.state`, using the token mechanism of sim.State should make this API obselete

Reply:
The issue is that `cloud.Service` requires `connectFn`, `disconnectFn`, and `messageFn` to be filled, but they are `inflight(str): void` variables

`sim.State.set` only accepts json.

 I attempted to parse them, but it resulted in errors.

Reply:
The `initialize` is necessary because I need the inflight functions to be defined before initializing the `cloud.Service`. Since these functions are configured in the preflight, it's not possible to use `sim.State.set` for configuration

And `sim.State.set` accepts `json`, not `inflight(str): void`.

Reply:
@eladcon open this issue about it

[captured field is undefined even when initialized](https://github.com/winglang/wing/issues/4324)

Reply:
this shouldn't be needed

Reply:
@eladb ,  it is required today because of https://github.com/winglang/wing/issues/4324. 
Do you have a workaround?

Reply:
yes, you are right.

another approach is to allow people to hook into the synthesis phase (e.g. override something like `onSynth()` to perform these types of finalization activities.

i think we need to fix #4324 *and* expose `onSynth()` to allow for both approaches.

Reply:
@marciocadev it is undocumented, but in the meantime, if you create a method called `_preSynthesize` it will be called just before synthesis.

[see playground](https://www.winglang.io/play/?code=YwBsAGEAcwBzACAARgBvAG8AIAB7AAoAIAAgAF8AcAByAGUAUwB5AG4AdABoAGUAcwBpAHoAZQAoACkAIAB7AAoAIAAgACAAIAB0AGgAcgBvAHcAIAAiAGgAZQBsAGwAbwAiADsACgAgACAAfQAKAH0ACgAKAG4AZQB3ACAARgBvAG8AKAApADsA)

i will create a PR to turn this into a public method that people can use.

Reply:
@eladb, I changed the `pub initialize()` to `_preSynthesize()` and it started giving me the following error when I try to use the simulator in the test

As in my test, I use Dynamo to capture who is logged in, it is not able to resolve the table at the pre-synthesis moment

![image](https://github.com/winglang/winglibs/assets/67694075/f3e72ece-4cec-4230-bfb2-68635ceb1264)

Reply:
i see... okay - keep the `initialize()` for now and i'll take a look at a proper solution.

thanks!